nf_core/subworkflows/info.py

```diff
@@ -0,0 +1,335 @@
+from rich.markdown import Markdown
+from rich.panel import Panel
+from rich.table import Table
+from rich.text import Text
+
+import nf_core.modules.module_utils
+from nf_core.modules.module_utils import get_repo_type
+from nf_core.modules.modules_json import ModulesJson
+from nf_core.modules.modules_repo import NF_CORE_MODULES_REMOTE, ModulesRepo
+
+log = logging.getLogger(__name__)
+
+
+class SubworkflowInfo(object):
+    """
+    Class to print information of a subworkflow.
+
+    Attributes
+    ----------
+    meta : YAML object
+        stores the information from meta.yml file
+    local_path : str
+        path of the local subworkflows
+    remote_location : str
+        remote repository URL
+    local : bool
+        indicates if the subworkflow is locally installed or not
+    repo_type : str
+        repository type. Can be either 'pipeline' or 'modules'
+    modules_json : ModulesJson object
+        contains 'modules.json' file information from a pipeline
+    module : str
+        name of the tool to get information from
+
+    Methods
+    -------
+    init_mod_name(subworkflow)
+        Makes sure that we have a subworkflow name
+    get_subworkflow_info()
+        Given the name of a subworkflow, parse meta.yml and print usage help
+    get_local_yaml()
+        Attempt to get the meta.yml file from a locally installed subworkflow
+    get_remote_yaml()
+        Attempt to get the meta.yml file from a remote repo
+    generate_subworkflow_info_help()
+        Take the parsed meta.yml and generate rich help
+    """
+
+    def __init__(self, pipeline_dir, tool, remote_url, branch, no_pull):
+        # super().__init__(pipeline_dir, remote_url, branch, no_pull)
+        self.dir = pipeline_dir
+        self.modules_repo = ModulesRepo(remote_url, branch, no_pull)
+        try:
+            if self.dir:
+                self.dir, self.repo_type = nf_core.modules.module_utils.get_repo_type(self.dir)
+            else:
+                self.repo_type = None
+        except LookupError as e:
+            raise UserWarning(e)
+
+        self.meta = None
+        self.local_path = None
+        self.remote_location = None
+        self.local = None
+
+        # Quietly check if this is a pipeline or not
+        if pipeline_dir:
+            try:
+                pipeline_dir, repo_type = get_repo_type(pipeline_dir, use_prompt=False)
+                log.debug(f"Found {repo_type} repo: {pipeline_dir}")
+            except UserWarning as e:
+                log.debug(f"Only showing remote info: {e}")
+                pipeline_dir = None
+
+        if self.repo_type == "pipeline":
+            # # Check modules directory structure
+            # self.check_modules_structure()
+            # Check modules.json up to date
+            self.modules_json = ModulesJson(self.dir)
+            self.modules_json.check_up_to_date()
+        else:
+            self.modules_json = None
+        self.module = self.init_mod_name(tool)
```

Comment:
I would change this variable name to `self.subworkflow` to make it clear. 
It will probably be modified once we implement the components class (see #1916) but meanwhile I think it can be confusing to name it module.